tests/test_dbos.py

```diff
@@ -36,6 +27,66 @@ def test_communicator(var: str) -> str:
         DBOS.logger.info("I'm test_communicator")
         return var
 
+    @dbos.workflow()
+    def test_workflow(var: str, var2: str) -> str:
+        DBOS.logger.info("I'm test_workflow")
+        if len(DBOS.parent_workflow_id):
+            DBOS.logger.info("  This is a child test_workflow")
+            # Note this assertion is only true if child wasn't assigned an ID explicitly
+            assert DBOS.workflow_id.startswith(DBOS.parent_workflow_id)
+        nonlocal wf_counter
+        wf_counter += 1
+        res = test_transaction(var2)
+        res2 = test_communicator(var)
+        return res + res2
+
+    @dbos.workflow()
+    def test_workflow_child() -> str:
+        nonlocal wf_counter
+        wf_counter += 1
+        res1 = test_workflow("child1", "child1")
+        return res1
+
+    wf_ac_counter: int = 0
+    txn_ac_counter: int = 0
+
+    @dbos.workflow()
+    def test_workflow_chilren() -> str:
+        nonlocal wf_counter
+        wf_counter += 1
+        res1 = test_workflow("child1", "child1")
+        wfh1 = dbos.start_workflow(test_workflow, "child2a", "child2a")
+        wfh2 = dbos.start_workflow(test_workflow, "child2b", "child2b")
+        res2 = wfh1.get_result()
+        res3 = wfh2.get_result()
+        return res1 + res2 + res3
+
+    @dbos.transaction()
+    def test_transaction_ac(var2: str) -> str:
+        rows = DBOS.sql_session.execute(sa.text("SELECT 1")).fetchall()
+        nonlocal txn_ac_counter
+        txn_ac_counter += 1
+        return var2 + str(rows[0][0])
+
+    @dbos.workflow()
+    def test_workflow_ac(var: str, var2: str) -> str:
+        DBOS.logger.info("I'm test_workflow assigned child id")
+        assert DBOS.workflow_id == "run_me_just_once"
+        res = test_transaction_ac(var2)
+        return var + res
+
+    @dbos.workflow()
+    def test_workflow_assignchild() -> str:
+        nonlocal wf_ac_counter
+        wf_ac_counter += 1
+        with SetWorkflowUUID("run_me_just_once"):
+            res1 = test_workflow_ac("child1", "child1")
+        with SetWorkflowUUID("run_me_just_once"):
+            # Use of "child2" on something that won't really run is deliberate to see if / where it appears
```

Comment:
I wonder if we want child2 to show up at all -- seems that we no longer validate the input to be the same across invocations. So it's possible to invoke a workflow with the same UUID but different inputs. We removed that check to reduce a DB read...

Reply:
What do you want - change it to child1 so the test doesn't touch on this controversial bit?

Reply:
Yeah I think let's just use child1 here.